lib/measured/unit.rb

```diff
@@ -40,6 +59,11 @@ def inverse_conversion_amount
 
   private
 
+  def with_case_sensitivity(comparison)
+    comparison = comparison.is_a?(Array) ? comparison.map{|c| c.downcase} : comparison.downcase if !case_sensitive
```

Comment:
Anytime you use `is_a?` it's a code smell. Ternary is unnecessary too.

How about something like?

```ruby
def with_case_sensitivity(comparison)
  comparison = [comparison].flatten.map(&:downcase) if !case_sensitive
  comparison
end
```